Doc/library/dis.rst

```diff
@@ -670,11 +668,45 @@ iterations of the loop.
    popped values are used to restore the exception state.
 
 
+.. opcode:: POP_FINALLY (preserve_tos)
+
+   Cleans up the value stack and the block stack.  If *preserve_tos* is not
```

Comment:
For the sake of static bytecode analyzers, perhaps it would be better to have two separate opcodes: one preserving TOS, the other not.

Reply:
Wouldn't this just complicate the code and the documentation?

Reply:
The code wouldn't be much more complicated (you can share the same implementation in the eval loop). OTOH I'm not sure it's necessary.

Other question: why are both POP_FINALLY and END_FINALLY needed? The documentation should be a bit clearer.

Reply:
`END_FINALLY` re-raises an exception, `POP_FINALLY 0` just drops it. `END_FINALLY` jumps to the return address, `POP_FINALLY 0` ignores it. And `POP_FINALLY 1` preserves TOS.

I don't know what to add to the documentation. Can you provide your changes?

If push 6 `None` instead of 1 `NULL` in `BEGIN_FINALLY`,  and push an exception block in `BEGIN_FINALLY`, and update other instructions accordingly, and add additional hacks to stack effect calculation, `POP_FINALLY 0` could be replaced with:
```
POP_TOP
POP_TOP
POP_TOP
POP_EXCEPT
```
and `POP_FINALLY 1` could be replaced with:
```
ROT_FOUR
POP_TOP
POP_TOP
POP_TOP
ROT_FOUR
POP_EXCEPT
```
This was my initial solution for @nascheme's examples. But this would add too much unneeded overhead and complication. The single new opcode has lesser cost.

Reply:
> END_FINALLY re-raises an exception, POP_FINALLY 0 just drops it. END_FINALLY jumps to the return address, POP_FINALLY 0 ignores it. 

Yes... So what I don't understand is why both are needed. In which case do you need to drop an exception?

Reply:
'break' and 'return' in 'finally' clause.

'continue' is specially prohibited, but it can be allowed too after merging this PR.

Reply:
Thank you. Can you add some explanation of this in the docs?